treeherder/seta/update_job_priority.py

```diff
@@ -0,0 +1,281 @@
+'''This module is used to add new jobs to the job priority table.
+
+This will query the Treeherder runnable api based on the latest task ID from
+mozilla-inbound's TaskCluster index.
+
+Known bug:
+ * Only considering mozilla-inbound makes SETA act similarly in all repositories where it is
+   active. Right now, this works for integration repositories since they tend to have
+   the same set of jobs. Unfortunately, this is less than ideal if we want to make this
+   work for project repositories.
+'''
+import copy
+import datetime
+import json
+import logging
+import os
+import time
+
+import requests
+from redo import retry
+
+from treeherder.seta.common import (db_map,
+                                    unique_key)
+from treeherder.seta.models import (DEFAULT_LOW_PRIORITY,
+                                    DEFAULT_TIMEOUT,
+                                    JobPriority)
+from treeherder.seta.runnable_jobs import RunnableJobs
+
+LOG = logging.getLogger(__name__)
+
+
+class ManageJobPriorityTable():
+    ''''Updates the seta_jobpriority table with information from new jobs'''
+    def __init__(self):
+        self.runnable_jobs = RunnableJobs()
+        self.data = None
+
+    def _unique_key(self, job):
+        """Return a key to query our uniqueness mapping system.
+
+        This makes sure that we use a consistent key between our code and selecting jobs from the
+        table.
+        """
+        return unique_key(testtype=str(job['testtype']),
+                          buildtype=str(job['platform_option']),
+                          platform=str(job['platform']))
+
+    def sanitized_data(self, runnable_jobs_data):
+        """We receive data from runnable jobs api and return the sanitized data that meets our needs.
+
+        This is a loop to remove duplicates (including buildsystem -> * transformations if needed)
+        By doing this, it allows us to have a single database query
+
+        It returns sanitized_list which will contain a subset which excludes:
+        * jobs that don't specify the platform
+        * jobs that don't specify the testtype
+        * if the job appears again, we replace build_system_type with '*'
```

Comment:
Maybe add: `(so if a job appears under both 'buildbot' and 'taskcluster', its build_system_type will be '*')`.